zipline/utils/security_list.py

```diff
@@ -0,0 +1,134 @@
+import os.path
+import pytz
+import pandas as pd
+from datetime import datetime
+from os import listdir
+
+DATE_FORMAT = "%Y%m%d"
+import zipline
+zipline_dir = os.path.join(*zipline.__path__)
+SECURITY_LISTS_DIR = os.path.join(zipline_dir, '..', 'security_lists')
+
+
+def loopback(symbol, *args, **kwargs):
+    return symbol
+
+
+class SecurityListSet(object):
+
+    def __init__(self, current_date_func, lookup_func=None):
+        if lookup_func is None:
+            self.lookup_func = loopback
+        else:
+            self.lookup_func = lookup_func
+        self.current_date_func = current_date_func
+        self._leveraged_etf = None
+
+    @property
+    def LEVERAGED_ETF_LIST(self):
+        if self._leveraged_etf is None:
+            self._leveraged_etf = SecurityList(
+                self.lookup_func,
+                load_from_directory('leveraged_etf_list'),
+                self.current_date_func
+            )
+        return self._leveraged_etf
+
+
+class SecurityList(object):
+
+    def __init__(self, lookup_func, data, current_date_func):
+        """
+        lookup_func: function that takes a string symbol and a date and
+        returns a Security object.
+        data: a nested dictionary:
+            knowledge_date -> lookup_date ->
+              {add: [symbol list], 'delete': []}, delete: [symbol list]}
+        current_date_func: function taking no parameters, returning
+            current datetime
+        """
+        self.lookup_func = lookup_func
+        self.data = data
+        self._cache = {}
+        self._knowledge_dates = self.make_knowledge_dates(self.data)
+        self.current_date = current_date_func
+        self.count = 0
+        self._list = set()
```

Comment:
I still think it is confusing that `self._list` is not a `list`, could we call this `_security_list` or something that is farther from the type name?